### PR TITLE
Post-BATB Tuning

### LINKED ISSUE #101: Tune flywheels speed up

### LINKED ISSUE #111: Abstract Auton Logic into Commands
This logic is repeated a lot (even repeated in preload and taxi) it might be smart to have a like command VisionDrivePathShoot that can be called multiple times with parameters because the steps are the same and this ensures we don't forget to do one.

_Originally posted by @TaylerUva in https://github.com/FRCTeam3255/2024_Offbot_Code/pull/105#discussion_r1797052627_

### LINKED ISSUE #118: Game piece in robot detection (not fully collected just entered robot)
beam break at the start of intake

### LINKED ISSUE #120: Adjust shuttle location

### LINKED ISSUE #121: Remove climber code

### LINKED ISSUE #124: Set flywheels to always on
Depending on power draw (seems like we aren't having any issues) we should have the flywheels always on to the high default as soon as we get a note

### PR DESCRIPTION
**Pull request by alicekuznetsov**
Closes #121 
Closes #101 
Closes #124 
Closes #111

### CODE REVIEW

**Comment by alicekuznetsov** [comment]
Lets finish #118 (needs the rumble still) in another pr

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -83,6 +85,7 @@ public static class constDrivetrain {
      * <p>
      * Observed maximum translational speed while manually driving on the
      * Competition Robot.
+     * S
```
Comment (TaylerUva) [inline src/main/java/frc/robot/Constants.java]: S

**Comment by alicekuznetsov** [comment]
guess who forgor to mark it as ready for review 😋

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 2]
```diff
@@ -61,49 +73,41 @@ public Shooter() {
...
-    leftConfig.Slot0.kD = prefShooter.leftShooterD;
-
-    leftConfig.MotionMagic.MotionMagicAcceleration = 400;
-    leftConfig.MotionMagic.MotionMagicJerk = 4000;
+    leftConfig.Slot0 = constShooter.LEFT_PID_SLOT_0_FAST;
+    leftConfig.Slot1 = constShooter.LEFT_PID_SLOT_1_SLOW;
+
+    leftConfig.MotionMagic.MotionMagicCruiseVelocity = 60;
+    leftConfig.MotionMagic.MotionMagicAcceleration = 600;
+    leftConfig.MotionMagic.MotionMagicJerk = 6000;
```
Comment (TaylerUva) [inline src/main/java/frc/robot/subsystems/Shooter.java:81]: How does motion magic work with slots? Can we do motion magic on a slot

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 3]
```diff
@@ -61,49 +73,41 @@ public Shooter() {
...
     pivotConfig.MotorOutput.Inverted = constShooter.PIVOT_INVERT;
     pivotConfig.MotorOutput.NeutralMode = constShooter.PIVOT_NEUTRAL_MODE;
-    pivotConfig.Slot0.kS = prefShooter.pivotShooterS;
-    pivotConfig.Slot0.kV = prefShooter.pivotShooterV;
-    pivotConfig.Slot0.kG = prefShooter.pivotShooterG;
-    pivotConfig.Slot0.kA = prefShooter.pivotShooterA;
-    pivotConfig.Slot0.kP = prefShooter.pivotShooterP;
-    pivotConfig.Slot0.kI = prefShooter.pivotShooterI;
-    pivotConfig.Slot0.kD = prefShooter.pivotShooterD;
     pivotConfig.Slot0.GravityType = constShooter.PIVOT_GRAVITY_TYPE;
```
Comment (TaylerUva) [inline src/main/java/frc/robot/subsystems/Shooter.java]: This should move in slot0 config. I think we are overriding it currently as setting slot0 occurs below

**Review by TaylerUva — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (alicekuznetsov) [reply src/main/java/frc/robot/Constants.java]: 🗣️ S 🗣️

File: src/main/java/frc/robot/subsystems/Shooter.java (hunk 2 above)
Comment (alicekuznetsov) [reply src/main/java/frc/robot/subsystems/Shooter.java:81]: Motion magic doesn't accept a slot as a parameter. It's settings are stored to the entire motor.
We can determine if we want to use motion magic when we create the request, as we need to create a `MotionMagicVoltage` object to use it. If you wanted to only use motion magic on a certain slot, you could just use a different request type whenever using that slot

**Review by TaylerUva — APPROVED** [review]

**Pull request merged**